tests/test_model.py

```diff
@@ -280,6 +280,88 @@ def setUp(self):
             rules = harness.model.get_rules("Enabled.Test")
             self.assertEqual(len(rules), 3)
 
+    async def test_mute_full_name(self):
+        """Test mute and unmute by full alarm name.
+        """
+        user = "test_mute_alarm"
+        remote_names = ["ScriptQueue:5", "Test:7"]
+
+        async with EnabledRulesHarness(names=remote_names) as harness:
+
+            harness.model.enable()
+            await harness.model.enable_task
+
+            full_rule_name = f"Enabled.{remote_names[0]}"
+            self.assertIn(full_rule_name, harness.model.rules)
+
+            for rule in harness.model.rules.values():
+                self.assertFalse(rule.alarm.muted)
+                self.assertEqual(rule.alarm.muted_severity, AlarmSeverity.NONE)
+                self.assertEqual(rule.alarm.muted_by, "")
+
+            # mute one rule by full name
+            harness.model.mute_alarm(name=full_rule_name,
+                                     duration=5,
+                                     severity=AlarmSeverity.WARNING,
+                                     user=user)
+            for name, rule in harness.model.rules.items():
+                if name == full_rule_name:
+                    self.assertTrue(rule.alarm.muted)
+                    self.assertEqual(rule.alarm.muted_severity, AlarmSeverity.WARNING)
+                    self.assertEqual(rule.alarm.muted_by, user)
+                else:
+                    self.assertFalse(rule.alarm.muted)
+                    self.assertEqual(rule.alarm.muted_severity, AlarmSeverity.NONE)
+                    self.assertEqual(rule.alarm.muted_by, "")
+
+            # unmute one rule by full name
+            harness.model.unmute_alarm(name=full_rule_name)
+            for rule in harness.model.rules.values():
```

Comment:
I feel like I've seen this "for rule in rules: all rules are muted" assert loop in quite a few places. Can you lift it into a testing function that can be used in multiple places? e.g. `check_all_rules_muted`

Reply:
Fair enough. I'll add `assertAlarmMuted` and `assertAlarmNotMuted` methods. That still leaves the loops gets rid of a lot of boilerplate.